INSTALL

```diff
@@ -380,3 +380,50 @@ We also have vcpkg ports for wolftpm, wolfmqtt and curl.
     resulting packages are placed in the root directory of the
     project.
 
+20. Building with xmss-reference lib for XMSS/XMSS^MT support [EXPERIMENTAL]
+
+    Experimental support for XMSS/XMSS^MT has been achieved by integration
+    with the xmss-reference implementation from RFC 8391 (XMSS: eXtended
+    Merkle Signature Scheme). We support a patched version of xmss-reference
+    based on this git commit:
+      171ccbd26f098542a67eb5d2b128281c80bd71a6
+    At the time of writing this, this is the HEAD of the master branch of
+    the xmss-reference project.
+
+    How to get the xmss-reference library:
+      $ mkdir ~/xmss
+      $ cd ~/xmss
+      $ git clone https://github.com/XMSS/xmss-reference.git src
+      $ cd src
+      $ git checkout 171ccbd26f098542a67eb5d2b128281c80bd71a6
+      $ git apply <path to xmss reference patch>
+
+    The patch may be found in the wolfssl-examples repo here:
+      pq/stateful_hash_sig/0001-Patch-to-support-xmss-reference-integration.patch
+
+    To build patched xmss-reference:
+      $ make xmss_lib.a
+
+    To build verify-only patched xmss-reference:
+      $ make xmss_verify_lib.a
+
+    Note that this patch adds wolfCrypt SHA256 hashing to xmss-reference, and
+    thus benefits from all the same asm speedups as wolfCrypt SHA hashing.
+    Depending on architecture you may build with --enable-intelasm, or
+    and --enable-armasm, and see 30-40% speedups in XMSS/XMSS^MT.
+
+    For full keygen, signing, verifying, and benchmarking support, build
+    wolfSSL with:
+      $ ./configure \
```

Comment:
wait...if xmss uses wolfssl for sha, then why doesn't it need to have wolfSSL around first? Is it done by registering callbacks?  If so, just a short sentence about  how its done would be nice.

Reply:
Correct, by registering a sha callback. Good point, I'll update these comments.